[BUG] Unable to create fabric_workspace resource from Terraform Cloud

### 🐛 What happened?

When working on a very simple proof of concept by creating a `fabric_workspace` resource, the plan failed to apply after about 4 min. The stack trace suggests its a bug and i submit it for  

Stack trace from the terraform-provider-fabric_v1.1.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xd4943c]

goroutine 81 [running]:
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.(*authErrorResponse).getErrFromResp(0xc000060480, 0x0)
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:275 +0x5c
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.(*ErrorHandler).processAuthFailedError(0x152d4c0?, {0x15370f0, 0xc0001e82a0}, 0xc00047e2d0)
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:214 +0x5d
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.(*ErrorHandler).processError(0xc000796930, {0x15370f0, 0xc0001e82a0}, {0x152d4c0, 0xc00003c100}, {0x0, 0x0}, {0x137998e, 0x10}, {0x1383f2f, ...})
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:136 +0x325
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.(*ErrorHandler).GetDiagsFromError(0xc0005aa7e0?, {0x15370f0, 0xc0001e82a0}, {0x152d4c0, 0xc00003c100}, {0x136f043?, 0x0?}, {0x0?, 0x0?})
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:89 +0x1d5
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.GetDiagsFromError(...)
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:307
github.com/microsoft/terraform-provider-fabric/internal/services/workspace.(*resourceWorkspace).Create(0xc00056c400, {0x1537048, 0xc0005aeb70}, {{{{0x1545890, 0xc0005afa10}, {0x11b9920, 0xc0005af7a0}}, {0x1548fb8, 0xc0004963c0}}, {{{0x1545890, ...}, ...}, ...}, ...}, ...)
github.com/microsoft/terraform-provider-fabric/internal/services/workspace/resource_workspace.go:123 +0x59c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0xc00024c908, {0x1537048, 0xc0005aeb70}, 0xc000797500, 0xc0007974d8)
github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_createresource.go:101 +0x577
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0xc00024c908, {0x1537048, 0xc0005aeb70}, 0xc000598140, 0xc0007975f0)
github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_applyresourcechange.go:57 +0x4aa
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0xc00024c908, {0x1537048?, 0xc0005aea80?}, 0xc0005980a0)
github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/proto6server/server_applyresourcechange.go:55 +0x385
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc00025be00, {0x1537048?, 0xc0007061b0?}, 0xc0001f0070)
github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:866 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x133d1e0, 0xc00025be00}, {0x1537048, 0xc0007061b0}, 0xc00072c000, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:611 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00025c200, {0x1537048, 0xc000706150}, 0xc00011e000, 0xc0003b9aa0, 0x1fa5918, 0x0)
google.golang.org/grpc@v1.69.4/server.go:1392 +0xfbe
google.golang.org/grpc.(*Server).handleStream(0xc00025c200, {0x15382a0, 0xc000416000}, 0xc00011e000)
google.golang.org/grpc@v1.69.4/server.go:1802 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
google.golang.org/grpc@v1.69.4/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 39
google.golang.org/grpc@v1.69.4/server.go:1041 +0x11d

Error: The terraform-provider-fabric_v1.1.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### 🔬 How to reproduce?

_No response_

### 🏗️ Code Sample / Log

resource "azurerm_resource_group" "rg-fabric" {
    name     = "rg-fabric"
    location = var.location
}

resource "azurerm_fabric_capacity" "tp-fabric-capacity-01" {
    name                = "tpfabriccapacity01"
    location            = azurerm_resource_group.rg-fabric.location
    resource_group_name = azurerm_resource_group.rg-fabric.name
    sku{
        name = var.sku
        tier = "Fabric"
    }

    administration_members = [var.admin_email]

    tags = {
        environment = "test"
    }
}

resource "fabric_workspace" "demo-workspace" {
    display_name        = "demo-workspace"
    description         = "this is a demo workspace"
    capacity_id         = "{my capacity id}"
}

### 📷 Screenshots

_No response_

### 📈 Expected behavior

_No response_

### 🌌 Environment (Provider Version)

1.1.0

### 🌌 Environment (Terraform Version)

1.12.0

### 🌌 Environment (OS)

Windows

### 📎 Additional context

Running this on Terraform Cloud

### 🔰 Code of Conduct

- [x] I agree to follow this project's Code of Conduct.

--- a/.changes/unreleased/fixed-20250515-194639.yaml
+++ b/.changes/unreleased/fixed-20250515-194639.yaml
@@ -1,0 +1,5 @@
+kind: fixed
+body: Handle runtime error / nil pointer in the AuthenticationFailedError error
+time: 2025-05-15T19:46:39.4887143-07:00
+custom:
+  Issue: "440"

--- a/internal/pkg/utils/errors.go
+++ b/internal/pkg/utils/errors.go
@@ -211,6 +211,11 @@ func (h *ErrorHandler) processFabricError(
 func (h *ErrorHandler) processAuthFailedError(ctx context.Context, errAuthFailed *azidentity.AuthenticationFailedError) (string, string) { //revive:disable-line:confusing-results
 	var errAuthResp authErrorResponse
 
+	// Check if RawResponse is nil to avoid panic
+	if errAuthFailed.RawResponse == nil {
+		return "AuthenticationFailedError", errAuthFailed.Error()
+	}
+
 	err := errAuthResp.getErrFromResp(errAuthFailed.RawResponse)
 	if err != nil {
 		return "Failed to parse authentication error response", err.Error()
@@ -272,7 +277,7 @@ type authErrorResponse struct {
 
 // getErrFromResp parses an HTTP response body into an authErrorResponse.
 func (e *authErrorResponse) getErrFromResp(resp *http.Response) error {
-	if resp.Body == nil {
+	if resp == nil || resp.Body == nil {
 		return nil
 	}
 

